### PR TITLE
UEFI-based SWUpdate and Secure Boot integration

### PR DESCRIPTION
**Pull request by jan-kiszka**
Not completely secure yet due to
 - missing hardening bits in SE Boot
 - missing security for persistent data
 - not yet shared tools for programming keys in to customer half of OTP

Still requires addition work on the documentation front, both to update it and to describe in more details how to adopt and customize these patterns for product images.

@gylstorffq @stormc

### CODE REVIEW

File: wic/iot2050-swu-secure.wks.in [hunk 1]
```diff
@@ -0,0 +1,24 @@
+#
+# Copyright (c) Siemens AG, 2021-2022
+#
+# Authors:
+#  Quirin Gylstorff <quirin.gylstorff@siemens.com>
```
Comment (gylstorffq) [inline wic/iot2050-swu-secure.wks.in:5]: Should be changed to the correct author.

File: recipes-bsp/patch-u-boot-env/files/patch-u-boot-env.service [hunk 2]
```diff
@@ -13,7 +13,7 @@ Before=swupdate.service shutdown.target
 
 [Service]
 Type=oneshot
-ExecStart=/usr/share/u-boot-env/patch-u-boot-env.sh sysselect /usr/share/u-boot-env/patch-u-boot-env.config
+ExecStart=sh -c '/usr/share/u-boot-env/patch-u-boot-env.sh watchdog_timeout_ms /usr/share/u-boot-env/patch-u-boot-env.config; fw_setenv watchdog_timeout_ms 60000'
```
Comment (gylstorffq) [inline recipes-bsp/patch-u-boot-env/files/patch-u-boot-env.service:16]: What is the reason for setting the variable `watchdog_timeout_ms` in the u-boot config script and the patch service?

Comment (jan-kiszka) [reply recipes-bsp/patch-u-boot-env/files/patch-u-boot-env.service:16]: The patch service can also handle current firmware 1.2.1 (for non-secure swupdate) by not only setting a non-zero timeout but also adding the related scripts that is missing in that version.

File: wic/iot2050-swu-secure.wks.in (hunk 1 above)
Comment (jan-kiszka) [reply wic/iot2050-swu-secure.wks.in:5]: Hmm, I felt like I derived it from something you wrote, but I might be right.

File: wic/iot2050-swu-secure.wks.in [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+#
+# This file is subject to the terms and conditions of the MIT License.  See
+# COPYING.MIT file in the top-level directory.
+#
+
+include ebg-signed-bootloader.inc
+
+# EFI Boot Guard environment/config partitions plus Kernel files
+part --source efibootguard-boot --size 32M --extra-space 0 --overhead-factor 1 --label BOOT0 --align 1024 --part-type=0700 --sourceparams "revision=2,signwith=/usr/bin/sign_secure_image.sh"
+part --source efibootguard-boot --size 32M --extra-space 0 --overhead-factor 1 --label BOOT1 --align 1024 --part-type=0700 --sourceparams "revision=1,signwith=/usr/bin/sign_secure_image.sh"
```
Comment (BaochengSu) [inline wic/iot2050-swu-secure.wks.in:15]: Why these boot partitions are not part of the included `ebg-signed-bootloader.inc`?

I notice that the boot partitions are part of the `ebg-sysparts.inc` for non-secure configuration.

Comment (jan-kiszka) [reply wic/iot2050-swu-secure.wks.in:15]: Good point, I guess I missed that imbalance while refactoring isar-cip-core. Let me quickly clean this up: https://gitlab.com/cip-project/cip-core/isar-cip-core/-/commit/71c708b27d2a0a6788639f3badeb5c13c20ff22f. Will be picked up here once merged.

**Comment by BaochengSu** [comment]
@jan-kiszka , Sorry but I could not recall the exact `missing hardening bits in SE Boot`, do you referring to the secure boot enable bit from SEBoot to SPL?

**Comment by jan-kiszka** [comment]
-> validated firewall settings

File: Kconfig [hunk 4]
```diff
@@ -99,6 +99,37 @@ config KAS_INCLUDE_LXDE
...
+	  root filesystem.
+
+	  This uses the public custMpk.key and certificate by default. Do not
+	  use this key in production, it is for demonstration purposes only.
+
+	  Note: The persistent /var and /home partitions are not yet protected!
+
+config SECURE_BOOT
+	bool "Secure boot"
+	depends on IMAGE_BOOT_PG1 || IMAGE_BOOT_PG2
```
Comment (BaochengSu) [inline Kconfig:118]: PG1 will not support secure boot. Should we need a disclaimer to record this information somewhere?

**Review by BaochengSu — APPROVED** [review]

**Pull request merged**